cmd/kubeadm/app/componentconfigs/kubelet.go

```diff
@@ -71,8 +72,14 @@ var kubeletHandler = handler{
 }
 
 func kubeletConfigFromCluster(h *handler, clientset clientset.Interface, clusterCfg *kubeadmapi.ClusterConfiguration) (kubeadmapi.ComponentConfig, error) {
+	// Parse and validate the version argument and resolve possible CI version labels
+	ver, err := kubeadmutil.KubernetesReleaseVersion(clusterCfg.KubernetesVersion)
```

Comment:
this is tricky.
it's related to https://github.com/kubernetes/kubeadm/issues/1582
for 1.25 the FG is GA and all the `k8sVersion` vars are not needed.
for older versions we need to normalize the version.

for older versions we can use something like `KubernetesReleaseVersion`
i'm assuming that `KubernetesReleaseVersion` will not fetch a version from the internet at this point in the runtime?

can't we instead just normalize the version with an existing function or manually extract `ci` / parse the ver here?

Reply:
a standard helper function that takes either e.g., `ci/v1.24.3-rc.0.29+99b7713ba36e8f` or `v1.24.3-rc.0.29+99b7713ba36e8f` as input and returns `v1.24.3-rc.0.29+99b7713ba36e8f` would be helpful here, we could simply invoke that function prior to doing semver parsing/validation

Reply:
> can't we instead just normalize the version with an existing function or manually extract ci / parse the ver here?

It makes sense.